listenbrainz/labs_api/labs/api/similar_recordings.py

```diff
@@ -27,28 +27,31 @@ def outputs(self):
         return None
 
     @staticmethod
-    def get_recordings_dataset(mb_curs, ts_curs, mbids, score_index=None, similar_mbid_index=None):
+    def get_recordings_dataset(mb_curs, ts_curs, mbids, score_idx=None, similar_mbid_idx=None):
+        """ Retrieve recording metadata for given list of mbids after resolving redirects, canonical redirects and
+        adding similarity data if available
+        """
         metadata = get_recordings_from_mbids(mb_curs, ts_curs, mbids)
         for r in metadata:
+            if score_idx and similar_mbid_idx:
+                similar_mbid = r["original_recording_mbid"]
+                r["score"] = score_idx[similar_mbid]
+                r["reference_mbid"] = similar_mbid_idx[similar_mbid]
+
             # remove unwanted fields from output
             r.pop("original_recording_mbid", None)
             r.pop("artist_credit_id", None)
             r.pop("length", None)
             r.pop("comment", None)
 
-            if score_index and similar_mbid_index:
-                similar_mbid = r["original_recording_mbid"]
-                r["score"] = score_index[similar_mbid]
-                r["reference_mbid"] = similar_mbid_index[similar_mbid]
-
         return {
             "type": "dataset",
             "columns": list(metadata[0].keys()),
             "data": metadata
         }
 
     def fetch(self, params, offset=-1, count=-1):
-        recording_mbids = [p["recording_mbid"] for p in params]
+        recording_mbids = [p["recording_mbid"].strip() for p in params]
```

Comment:
Yes, we need this here, but I think we need to add it to dataset hoster. :)

Reply:
Makes sense.